bindings/jaxrs/src/main/java/org/commonjava/indy/core/bind/jaxrs/ContentAccessHandler.java

```diff
@@ -172,6 +180,58 @@ public Response doDelete( final String packageType, final String type, final Str
         return doDelete( packageType, type, name, path, eventMetadata, null );
     }
 
+    public Response doDelete( final String packageType, final String type, final String name,
+                              final HttpServletRequest request, EventMetadata eventMetadata )
+    {
+        setContext( PACKAGE_TYPE, packageType );
+
+        if ( !PackageTypes.contains( packageType ) )
+        {
+            ResponseBuilder builder = Response.status( 400 );
+            return builder.build();
+        }
+
+        Set<String> paths;
+        try
+        {
+            paths = mapper.readValue( request.getInputStream(), new TypeReference<Set<String>>() {} );
```

Comment:
Rather than a general HttpServletRequest, I'd prefer to use a concrete request object, e.g., DemoteRequest. Using the concrete req object, and the resteasy will ser/deser it automatically. It will be easy for both client and server code.

Reply:
Yeah, concrete one looks better and easy to extend.